devito/__init__.py

```diff
@@ -101,8 +101,21 @@ def reinit_compiler(val):
 # optimisations.
 configuration.add('safe-math', 0, [0, 1], preprocessor=bool, callback=reinit_compiler)
 
+
 # Enable/disable automatic padding for allocated data
-configuration.add('autopadding', False, [False, True])
+def _preprocess_autopadding(v):
+    return {
+        '0': False,
```

Comment:
Does this also need a `{False: False}` mapping?

Reply:
no

```
>>> False == 0
True
>>> hash(False) == hash(0)
True
```